products/sle/main.pm

```diff
@@ -35,6 +35,138 @@ use known_bugs;
 
 init_main();
 
+package SchedulerFactory{
+    use Data::Dumper ;    
+    
+    sub new {
+	my ($class, $testname) = @_;
+	#print "attrs --> $attrs{'testname'}";
+	my $self = { 
+	    testname_factory => TestSuiteFactory::getTestSuite($testname)
+	};
+	bless $self, $class;
+	return $self;    
+    };
+    
+    sub createScheduler {
+	my ($self) = shift;
+	$self->{testname_factory};
+    }
+}
+
+
+package TestSuiteFactory {
+    my %sleTests = (
+	boot_linuxrc => sub { Boot_linuxrc->new('boot_linuxrc')},
+	installer_extended_textmode => sub {InstallerExtendedTextmode->new('installer_extended_textmode')}
+	);
+    sub getTestSuite{
+	my ($name) = shift;
+	print "name is $name\n";
+	return $sleTests{$name}->()
+    }
+}
+
+package TestSuiteInterface {
+    use File::Find;
+    use File::Basename;
+    use main_common;
+    BEGIN {
+	unshift @INC, dirname(__FILE__) . '/../../lib';
+    }
+    sub new {
+	my ($class, $name) = @_;
+	my $self = {};
+	$self->{name} = $name;
+	bless $self, $class;
+	return $self;
+    }
+    sub getVariables {
+	my ($self) = shift;
+	return $self->{variables}
+    }
+    sub loadtests {
+	my ($self) = shift;
+	{
+	    #die "oops" unless $self->{scheduler}
+	    $_->() for (values %{$self->{scheduler}});
+	    return 1;
+	}
+	return 0;
+    }
+}
+
+package InstallerExtendedTextmode {
```

Comment:
you already cut this piece logically from main.pm , so why not go further and cut it physically ( put into separate file ) ?

Reply:
Already done. I was keeping the files in the main.pm for convenience but all is in the separate files in the last commit